tests/integration/test_lockfile.py

```diff
@@ -0,0 +1,63 @@
+from collections import defaultdict
+import json
+import pytest
+
+from pipenv.project import Project
+from pipenv.utils import requirements
+
+
+@pytest.fixture
+def pypi_lockfile():
+    lockfile = defaultdict(dict)
+    lockfile["_meta"] = {
+        "sources": [
+            {
+                "name": "pypi",
+                "url": "https://pypi.org/simple",
+                "verify_ssl": True,
+            }
+        ]
+    }
+    yield lockfile
+
+
+def test_git_branch_contains_slashes(
+        pipenv_instance_pypi, pypi_lockfile
+):
+    pypi_lockfile["default"]["google-api-python-client"] = {
+        "git": "https://github.com/thehesiod/google-api-python-client.git@thehesiod/batch-retries2",
+        "markers": "python_version >= '3.7'",
```

Comment:
Will this test break if the branch is removed?
We should have an isolated test that does not depend on other people's repository.

Reply:
These are unit level tests but I had to do them as integration to leverage the pipfile/lockfile stuff we get from those fixtures, but I am starting to write tests that are more unit level in nature -- so we are really just testing the string conversions that happen through these constructs -- no actual lock resolution or installation happening in these tests.  More like is it giving intelligible things to pip resolver or installer based on the inputs.  For example, that commit hash was from a different test and not actually part of that repo.

Reply:
I don't care using mine, however I'm guessing you'll want to keep your own long term ;)

Reply:
> These are unit level tests but I had to do them as integration to leverage the pipfile/lockfile stuff we get from those fixtures, but I am starting to write tests that are more unit level in nature -- so we are really just testing the string conversions that happen through these constructs -- no actual lock resolution or installation happening in these tests. More like is it giving intelligible things to pip resolver or installer based on the inputs. For example, that commit hash was from a different test and not actually part of that repo.

You also submit a PR with the example files to plette repo. I started a collection of test cases there.